packages/app/src/AppWrapper.js

```diff
@@ -42,15 +48,46 @@ const AppWrapper = () => {
     }, [])
 
     const schemas = ['visualization', 'organisationUnit', 'userGroup']
+    const d2Config = {
+        schemas,
+    }
 
     return (
-        <Provider store={store}>
-            <D2Shim schemas={schemas}>
-                {params => (
-                    <App {...params} dataEngine={engine} ouLevels={ouLevels} />
-                )}
-            </D2Shim>
-        </Provider>
+        <ReduxProvider store={store}>
+            <UserSettingsProvider>
+                <UserSettingsCtx.Consumer>
+                    {({ userSettings }) => {
+                        return userSettings?.keyUiLocale ? (
+                            <D2Shim
+                                d2Config={d2Config}
+                                i18nRoot="./i18n-old"
+                                locale={userSettings.keyUiLocale}
+                            >
+                                {({ d2 }) => {
+                                    if (!d2) {
+                                        // TODO: Handle errors in d2 initialization
+                                        return null
+                                    } else {
+                                        return (
+                                            <App
+                                                d2={d2}
+                                                location={history.location}
+                                                baseUrl={baseUrl}
```

Comment:
Could the App component just use the `useConfig` hook directly or is it not a functional component?